lib/modules/manager/pip-compile/extract.ts

```diff
@@ -200,6 +205,25 @@ export async function extractAllPackageFiles(
     depsBetweenFiles,
     packageFiles,
   );
+
+  // This needs to go in reverse order to handle transitive dependencies
+  for (const packageFile of [...result].reverse()) {
+    for (const reqFile of packageFile.managerData?.requirementsFiles ?? []) {
+      let sourceFile: PackageFile | undefined = undefined;
+      if (fileMatches.includes(reqFile)) {
+        sourceFile = lockFileSources.get(reqFile);
+      } else if (packageFiles.has(reqFile)) {
+        sourceFile = packageFiles.get(reqFile);
+      }
+      if (!sourceFile) {
+        logger.warn(
+          `pip-compile: ${packageFile.packageFile} references ${reqFile} which does not appear to be a requirements file managed by pip-compile`,
+        );
+        continue;
+      }
+      sourceFile.lockFiles!.push(...packageFile.lockFiles!);
```

Comment:
What if a lock file is duplicated across package files before this operation?

Reply:
Hmm, I suppose that could cause a problem with something like this
```mermaid
graph LR
    a.in --> b.in
    a.in --> c.in
    b.in --> d.in
    c.in --> d.in
```

In that case `a.in`'s `lockFiles` would be `["a.txt", "b.txt", "d.txt", "c.txt", "d.txt"]` and Renovate would end up running
```
pip-compile a.in
pip-compile b.in
pip-compile d.in
pip-compile c.in
pip-compile d.in
``

Where the first instance of `pip-compile d.in` would likely fail since `b.txt` and `c.txt` could ask for different versions of a newly-updated package.

I'll look into fixing that on Monday.